test/dshell/dll.d

```diff
@@ -5,6 +5,12 @@ int main()
     version (Windows) if (Vars.MODEL == "32omf") // Avoid optlink
         return DISABLED;
 
+    version (OSX) // Shared libraries are not yet supported on OSX
```

Comment:
If this is a DMD only problem, this should probably be `version(DigitalMars) version (OSX)` rather that just `version(OSX)`

Reply:
Since GDC resolves it with the @weak attribute, I agree with @thewilsonator

Reply:
I don't run the dshell tests, does ldc?

Reply:
I presume  we do, cc @kinke?

Reply:
Yep, we do, and this test works with LDC on macOS, with a little adaptation (https://github.com/ldc-developers/dmd-testsuite/blob/04d6f174b71032ea2e7ba96db81ed136e6eb1329/dshell/dll.d#L27-L28).